### PR TITLE
Implement eth_getBlockReceipts API

### PR DESCRIPTION
**Pull request by khoau2u**
Please check if what you want to add to `go-u2u` list meets [Contribution guidelines](https://github.com/unicornultrafoundation/go-u2u/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/unicornultrafoundation/go-u2u/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/unicornultrafoundation/go-u2u/blob/master/CONTRIBUTING.md#quality-standard).

- References: This PR refer to issue number no
- Description: #106 
- [x] API definition and logic
- [ ] Test suite

### CODE REVIEW

File: ethapi/api.go [hunk 1]
```diff
@@ -1806,13 +1847,7 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 		"logs":              receipt.Logs,
 		"logsBloom":         &receipt.Bloom,
 		"type":              hexutil.Uint(tx.Type()),
-	}
-	// Assign the effective gas price paid
-	if header.BaseFee == nil {
-		fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
-	} else {
-		gasPrice := new(big.Int).Add(header.BaseFee, tx.EffectiveGasTipValue(header.BaseFee))
-		fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
```
Comment (trinhdn97) [inline ethapi/api.go:1815]: This should be how we calculate `effectiveGasPrice` = `baseFee` + `effectiveGasTip` for dynamic fee txs

**Review by trinhdn97 — CHANGES_REQUESTED** [review]

File: ethapi/api.go (hunk 1 above)
Comment (khoau2u) [reply ethapi/api.go:1815]: First statement is for pre-1559, second is for 1559 and on ward

Comment (trinhdn97) [reply ethapi/api.go:1815]: I.e you deleted this and used pre-1559 rule for all txs

File: ethapi/api.go [hunk 2]
```diff
@@ -1791,13 +1822,23 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
...
+}
+
+// marshalReceipt marshals a transaction receipt into a JSON object.
+func marshalReceipt(receipt *types.Receipt, signer types.Signer, tx *types.Transaction, txIndex uint64, baseFee *big.Int) map[string]interface{} {
+	from, _ := types.Sender(signer, tx)
+	// Assign the effective gas price paid
+	gasPrice := tx.GasPrice()
+	if baseFee != nil {
+		gasPrice = big.NewInt(0).Add(baseFee, tx.EffectiveGasTipValue(baseFee))
+	}
```
Comment (khoau2u) [inline ethapi/api.go:1836]: It just refactored, gas price logic actually is not removed

File: ethapi/api.go (hunk 1 above)
Comment (khoau2u) [reply ethapi/api.go:1815]: Logic actually moved up one block, not removed or wrong

**Pull request merged**